bot/interact.go

```diff
@@ -108,6 +124,63 @@ func interactAPIHandler(botContext cmd.BotContext) http.HandlerFunc {
 					Value:     1,
 					Timestamp: time.Now(),
 				})
+			case "delete_thread":
+				// We early set the Content-Type header for any response. This is important.
+				w.Header().Set("Content-Type", "application/json")
+
+				if !botContext.IsStaff(message.User.ID) {
+					_ = json.NewEncoder(w).Encode(
+						slack.NewErrorsViewSubmissionResponse(map[string]string{"input_block": "You are not allowed to delete threads. Please contact any Staff member."}),
+					)
+					return
+				}
+
+				messageData := strings.Split(strings.Trim(message.View.PrivateMetadata, `"`), "|") // For some reason, slack adds an extra double quote
+				channelID := messageData[0]
+				messageTS := messageData[1]
+
+				repliesParams := &slack.GetConversationRepliesParameters{
+					ChannelID: channelID,
+					Timestamp: messageTS,
+				}
+
+				var threadMessages []slack.Message
+				var cursor = ""
+				var more = true
+
+				for more {
+					repliesParams.Cursor = cursor
+					var replies []slack.Message
+					replies, more, cursor, err = botContext.Client.GetConversationReplies(repliesParams)
+					if err != nil {
+						_ = json.NewEncoder(w).Encode(slack.NewErrorsViewSubmissionResponse(map[string]string{"input_block": err.Error()}))
+						return
+					}
+
+					threadMessages = append(threadMessages, replies...)
+				}
+
+				go func() {
+					ok := deleteThreadMessages(botContext, threadMessages, channelID, messageTS)
+					if !ok {
+						log.Println("Thread deletion finished with errors (see logs)", message.View.PrivateMetadata)
+					} else {
+						log.Println("Thread deletion finished successfully", message.View.PrivateMetadata)
+					}
+
+					//Sending metrics
+					botContext.Harvester.RecordMetric(telemetry.Count{
+						Name: "candebot.thread.deleted",
+						Attributes: map[string]interface{}{
+							"candebot_version": botContext.Version,
+							"errored":          !ok,
```

Comment:
maybe we could add the number of messages deleted

Reply:
Done!